resources/Microsoft.Windows.Setting.Language/Microsoft.Windows.Setting.Language.psm1

```diff
@@ -0,0 +1,213 @@
+    # helpful for users to discover what packages can be installed
+    $allLangues = [System.Globalization.CultureInfo]::GetCultures('AllCultures')
+    foreach ($culture in $allLangues) {
+        if ($out.LocaleName -notcontains $culture.Name -and -not ([string]::IsNullOrEmpty($culture.Name))) {
+            $language = [Language]::new($culture.Name, $false)
+            $out.Add($language)
+        }
+    }
+
+    return $out
+}
+#endregion Functions
+
+#region Classes
+<#
+.SYNOPSIS
+    The `Language` DSC Resource allows you to install, update, and uninstall languages on your local Windows machine.
+
+.PARAMETER LocaleName
+    The name of the language. This is the language tag that represents the language. For example, `en-US` represents English (United States).
+    To get a full list of languages available, use the `Get-LocaleList` function or Export() method.
+
+.PARAMETER Exist
+    Indicates whether the package should exist. Defaults to $true.
+
+.EXAMPLE
+    PS C:\> Invoke-DscResource -ModuleName Microsoft.Windows.Setting.Language -Name Language -Method Set -Property @{ LocaleName = 'en-US' }
+
+    This example installs the English (United States) language on the local machine.
+#>
+[DscResource()]
+class Language {
+    [DscProperty(Key)]
+    [string] $LocaleName
+
+    [DscProperty()]
+    [bool] $Exist = $true
+
+    static [hashtable] $InstalledLocality
+
+    Language() {
+        [Language]::GetInstalledLocality()
+    }
+
+    Language([string] $LocaleName, [bool] $Exist) {
+        $this.LocaleName = $LocaleName
+        $this.Exist = $Exist
+    }
+
+    [Language] Get() {
+        $keyExist = [Language]::InstalledLocality.ContainsKey(($this.LocaleName))
+
+        $currentState = [Language]::InstalledLocality[$this.LocaleName]
+
+        if (-not $keyExist) {
+            return [Language]::new($this.LocaleName, $false)
+        }
+        
+        return $currentState
+    }
+
+    [void] Set() {
+        if ($this.Test()) {
+            return
+        }
+
+        if ($this.Exist) {
+            # use the LanguagePackManagement module to install the language (requires elevation). International does not have a cmdlet to install language
+            Install-Language -Language $this.LocaleName
+        } else {
+            Uninstall-Language -Language $this.LocaleName
+        }
+    }
+
+    [bool] Test() {
+        $currentState = $this.Get()
+
+        if ($currentState.Exist -ne $this.Exist) {
+            return $false
+        }
+        
+        return $true
+    }
+
+    static [Language[]] Export() {
+        return Get-LocaleList
+    }
+
+    #region Language helper functions
+    static [void] GetInstalledLocality() {   
+        [Language]::InstalledLocality = @{}
+
+        foreach ($locality in [Language]::Export()) {
+            [Language]::InstalledLocality[$locality.LocaleName] = $locality
+        }
+    }
+    #endRegion Language helper functions
```

Comment:
This helper function is slightly confusing for me. Why not just use the function in export and create the installed map in the constructor?